lib/tidewave/mcp/server.ex

```diff
@@ -48,7 +49,10 @@ defmodule Tidewave.MCP.Server do
         true
     end
 
-    for tool <- tools, listable?.(tool) do
+    for tool <- tools,
+        listable?.(tool),
+        is_nil(assigns[:include_tools]) or tool.name in assigns[:include_tools],
```

Comment:
I would love to be precise here and make sure this is always set.

```suggestion
        is_nil(assigns.include_tools) or tool.name in assigns.include_tools,
```